.travis.yml

```diff
@@ -92,9 +100,16 @@ install:
 before_script:
   - if [ "$BROWSER_NAME" = "chrome" ]; then mkdir chromedriver; wget -q -t 3 https://chromedriver.storage.googleapis.com/$CHROMEDRIVER_VERSION/chromedriver_linux64.zip; unzip chromedriver_linux64 -d chromedriver; fi
   - if [ "$BROWSER_NAME" = "chrome" ]; then export CHROMEDRIVER_PATH=$PWD/chromedriver/chromedriver; fi
+  - if [ "$GECKODRIVER" = "1" ]; then mkdir geckodriver; wget -q -t 3 https://github.com/mozilla/geckodriver/releases/download/v0.22.0/geckodriver-v0.22.0-linux64.tar.gz; tar xzf geckodriver-v0.22.0-linux64.tar.gz -C geckodriver; fi
   - sh -e /etc/init.d/xvfb start
-  - if [ ! -f jar/selenium-server-standalone-3.8.1.jar ]; then wget -q -t 3 -P jar https://selenium-release.storage.googleapis.com/3.8/selenium-server-standalone-3.8.1.jar; fi
-  - java -Dwebdriver.firefox.marionette=false -Dwebdriver.chrome.driver="$CHROMEDRIVER_PATH" -jar jar/selenium-server-standalone-3.8.1.jar -enablePassThrough false -log ./logs/selenium.log &
+  - if [ ! -f jar/selenium-server-standalone-3.14.0.jar ]; then wget -q -t 3 -P jar https://selenium-release.storage.googleapis.com/3.14/selenium-server-standalone-3.14.0.jar; fi
+  - |
+    if [ "$GECKODRIVER" = "1" ]
+    then
+      geckodriver/geckodriver &
```

Comment:
why not running it through the selenium server like others ? You only need to remove the `webdriver.firefox.marionette=false` option you are adding, to go back to the default behavior (using geckdriver for firefox)

Reply:
Because it doesn’t behave the same way through Selenium, even with this flag removed.

I’ve no clue why, maybe that Selenium server add some kind of compatibility layer of its own.

Reply:
then, should we have another job running tests through Selenium but with a geckodriver-backed Firefox ? IMO, that's also a common use case (the legacy FirefoxDriver is probably seldom-used today, as it is incompatible with any Firefox version released since months).